### PR TITLE
LUCENE-10459: add demo dependencies to third party modules. Add an IT…

### PR DESCRIPTION
**Pull request by dweiss**

### CODE REVIEW

File: dev-tools/scripts/smokeTestRelease.py [hunk 1]
```diff
@@ -657,8 +657,8 @@ def testDemo(run_java, isSrc, version, jdk):
     indexFilesCmd = 'java -cp "%s" -Dsmoketester=true org.apache.lucene.demo.IndexFiles -index index -docs %s' % (cp, docsDir)
     searchFilesCmd = 'java -cp "%s" org.apache.lucene.demo.SearchFiles -index index -query lucene' % cp
   else:
-    # For binary release, set up classpath as modules.
-    cp = "--module-path modules"
+    # For binary release, set up module path.
+    cp = "--module-path %s" % (sep.join("modules", "modules-thirdparty"))
```
Comment (mocobeta) [inline dev-tools/scripts/smokeTestRelease.py:661]: `join()` takes a list;  `[` and `]` is needed.
```suggestion
    cp = "--module-path %s" % (sep.join(["modules", "modules-thirdparty"]))
```

Comment (mocobeta) [reply dev-tools/scripts/smokeTestRelease.py:661]: It is a trivial syntax correction (I confirmed it works). Let me commit this.

**Comment by mocobeta** [comment]
I just corrected syntax and cherry-picked some changes in #739 to make the smoke tester work.

**Comment by mocobeta** [comment]
I cherry-picked these commits on my branch_9_1 and ran the smoke tester.
```
lucene $ /usr/bin/python3 -u dev-tools/scripts/smokeTestRelease.py --not-signed --tmp-dir /mnt/hdd/tmp/smoketester file:///tmp/lucene-9.1/lucene-9.1.0-RC1-rev-b93d62ba57532a4d523e3a4a9579b0e9e47bf793

Java 11 JAVA_HOME=/home/moco/.sdkman/candidates/java/11.0.12-open
NOTE: output encoding is utf-8

Load release URL "file:///tmp/lucene-9.1/lucene-9.1.0-RC1-rev-b93d62ba57532a4d523e3a4a9579b0e9e47bf793"...

Get KEYS...
    Downloading online KEYS file https://archive.apache.org/dist/lucene/KEYS
Downloading https://archive.apache.org/dist/lucene/KEYS
    0.3 MB in 1.75 sec (0.2 MB/sec)

Test Lucene...
  test basics...
  check changes HTML...
  download lucene-9.1.0-src.tgz...
Downloading file:///tmp/lucene-9.1/lucene-9.1.0-RC1-rev-b93d62ba57532a4d523e3a4a9579b0e9e47bf793/lucene/lucene-9.1.0-src.tgz
    47.3 MB in 0.03 sec (1601.8 MB/sec)
    verify sha512 digest
  download lucene-9.1.0.tgz...
Downloading file:///tmp/lucene-9.1/lucene-9.1.0-RC1-rev-b93d62ba57532a4d523e3a4a9579b0e9e47bf793/lucene/lucene-9.1.0.tgz
    64.8 MB in 0.04 sec (1535.7 MB/sec)
    verify sha512 digest
  unpack lucene-9.1.0.tgz...
  lucene-9.1.0.tgz
    Checking LICENSE
    Checking NOTICE
    Checking README
    Checking JRE_VERSION_MIGRATION
    Checking CHANGES
    Checking MIGRATE
    Checking SYSTEM_REQUIREMENTS
    Checking bin
    Checking docs
    Checking licenses
    Checking modules
    Checking modules-thirdparty
    Checking modules-test-framework
    verify JAR metadata/identity/no javax.* or java.* classes...
    test demo with 11...
      got 4167 hits for query "lucene"
    checkindex with 11...
  unpack lucene-9.1.0-src.tgz...
  lucene-9.1.0-src.tgz
    Checking LICENSE
    Checking NOTICE
    Checking README
    Checking JRE_VERSION_MIGRATION
    Checking CHANGES
    Checking MIGRATE
    Checking SYSTEM_REQUIREMENTS
    Checking build.gradle
    Checking buildSrc
    Checking dev-docs
    Checking dev-tools
    Checking gradle
    Checking gradlew
    Checking gradlew.bat
    Checking help
    Checking lucene
    Checking settings.gradle
    Checking versions.lock
    Checking versions.props
    Checking analysis
    Checking analysis.tests
    Checking backward-codecs
    Checking benchmark
    Checking classification
    Checking codecs
    Checking core
    Checking core.tests
    Checking distribution.tests
    Checking demo
    Checking expressions
    Checking facet
    Checking grouping
    Checking highlighter
    Checking join
    Checking luke
    Checking memory
    Checking misc
    Checking monitor
    Checking queries
    Checking queryparser
    Checking replicator
    Checking sandbox
    Checking spatial-extras
    Checking spatial-test-fixtures
    Checking spatial3d
    Checking suggest
    Checking test-framework
    Checking licenses
    Checking build.gradle
    Checking documentation
    Checking distribution
    Checking dev-docs
    make sure no JARs/WARs in src dist...
    run "./gradlew --no-daemon check -p lucene/documentation"
    run tests w/ Java 11 and testArgs='-Dtests.nightly=true -Dtests.badapples=false '...
    compile jars w/ Java 11
    test demo with 11...
      got 230 hits for query "lucene"
    checkindex with 11...
  confirm all releases have coverage in TestBackwardsCompatibility
    find all past Lucene releases...
    run TestBackwardsCompatibility..
    success!

Test Maven artifacts...
    download artifacts
................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
    verify that each binary artifact has a deployed POM...
    verify Maven artifacts' md5/sha1 digests...
    check for javadoc and sources artifacts...
    verify deployed POMs' coordinates...
    unpack lucene-9.1.0.tgz...
    verify that Maven artifacts are same as in the binary distribution...
    verify JAR metadata/identity/no javax.* or java.* classes...

SUCCESS! [0:47:03.692942]
```

**Comment by dweiss** [comment]
Thanks @mocobeta !

**Review by jtibshirani — APPROVED** [review]
These changes make sense to me too, I guess #739 wasn't as hacky as I thought :)

**Pull request merged**